test/sql/cast/string_to_nested_types_cast.test_slow

```diff
@@ -78,7 +78,7 @@ SELECT CAST(LIST(timestamp_ns)::VARCHAR AS TIME[]) FROM test_all_types();
 query I
 SELECT CAST(LIST(blob)::VARCHAR AS BLOB[]) FROM test_all_types();
 ----
-[thisisalongblob\x00withnullbytes, \x00\x00\x00a, NULL]
+[thisisalongblobx00withnullbytes, x00x00x00a, NULL]
```

Comment:
Let's make backslash only work as an escape character if it is used within a quoted string